### PR TITLE
fix(profiles): Pass `homeDir, projectDir` when creating `ProfileInfo` instance

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Fixed a bug where a new `ProfileInfo` instance was being created when initializing Zowe profiles, but that instance was missing the `homeDir` and `projectDir` from its constructor options.

## Release Notes

Milestone: 3.0.1

Changelog:

- Fixed an issue where Zowe Explorer displayed a "No Zowe client configurations" prompt when a project user configuration existed but no global configuration was present. [#3168](https://github.com/zowe/zowe-explorer-vscode/issues/3168)
- Fixed an issue where the `ProfilesUtils.getProfileInfo` function returned a new `ProfileInfo` instance that ignored the `ZOWE_CLI_HOME` environment variable and workspace paths. [#3168](https://github.com/zowe/zowe-explorer-vscode/issues/3168)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [ ] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [x] These changes may need ported to the appropriate branches (list here): `v2-lts`

### CODE REVIEW

**Review by zFernand0 — APPROVED** [review]

File: packages/zowe-explorer/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,9 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 
 ### Bug fixes
 
+- Fixed an issue where Zowe Explorer displayed a "No Zowe client configurations" prompt when a project, user-level config exists but no global config was present. [#3168](https://github.com/zowe/zowe-explorer-vscode/issues/3168)
```
Comment (anaxceron) [inline packages/zowe-explorer/CHANGELOG.md:11]: what do we mean by "user-level"? If we mean a user config file, can we reword this to something like "... prompt when a project user configuration file existed but no global configuration was present."

also, please note the verb tenses above.

File: packages/zowe-explorer/CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,9 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
...
+- Fixed an issue where the `ProfilesUtils.getProfileInfo` function returned a new `ProfileInfo` instance that did not respect the `ZOWE_CLI_HOME` environment variable and workspace paths. [#3168](https://github.com/zowe/zowe-explorer-vscode/issues/3168)
```
Comment (anaxceron) [inline packages/zowe-explorer/CHANGELOG.md:12]: what do we mean by "did not respect"? Did the instance simply ignore the variable and workspace path? If so, can we use "ignore" instead?

**Review by anaxceron — COMMENTED** [review]
Left a couple comments re: changelog.

File: packages/zowe-explorer/CHANGELOG.md (hunk 1 above)
Comment (traeok) [reply packages/zowe-explorer/CHANGELOG.md:11]: Thanks, I adjusted the verb tenses and updated the entry 👍

File: packages/zowe-explorer/CHANGELOG.md (hunk 2 above)
Comment (traeok) [reply packages/zowe-explorer/CHANGELOG.md:12]: That's correct - updated this entry as well

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @traeok!

**Review by zFernand0 — APPROVED** [review]
